ssh/tailssh/incubator.go

```diff
@@ -780,6 +785,20 @@ func setGroups(groupIDs []int) error {
 		// this to work for more things than it previously did.
 		groupIDs = groupIDs[:16]
 	}
+	if runtime.GOOS == "freebsd" {
+		// On FreeBSD, the first entry returned from the getgroups(2)
+		// syscall is the egid, and changing it with setgroups(2)
+		// changes the egid of the process. This is technically a
+		// violation of the POSIX standard; see the following article
+		// for more detail:
+		//    https://www.usenix.org/system/files/login/articles/325-tsafrir.pdf
+		//
+		// In this case, we add an entry at the beginning of the
+		// groupIDs list containing the current egid, which replicates
+		// the behaviour on every other platform of just changing the
+		// supplementary groups and not the egid.
+		groupIDs = append([]int{os.Getegid()}, groupIDs...)
```

Comment:
Do you want to filter out egid from the groupIDs?

Reply:
Surprisingly not; FreeBSD can legitimately return an array of `[]int{0, 0, 1234}` from `getgroups`, and happily accepts an array with duplicates passed to `setgroups`. The linked article also does not filter out egid in their example code.